From 890edaefc84a2596a3f17d1e9967935a1984e302 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:32 -0400
Subject: [virt] kvm: expose SMAP feature to guest

Message-id: <1406040016-3289-178-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86256
O-Subject: [RHEL7 PATCH v2 177/221] KVM: expose SMAP feature to guest
Bugzilla: 1084616 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Feng Wu <feng.wu@intel.com>

This patch exposes SMAP feature to guest

Signed-off-by: Feng Wu <feng.wu@intel.com>
Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
(cherry picked from commit de935ae15b0db57f7fb837b37380ee94d4b0fec3)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index 4ca7d97..333b88d 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -308,7 +308,7 @@ static inline int __do_cpuid_ent(struct kvm_cpuid_entry2 *entry, u32 function,
  const u32 kvm_supported_word9_x86_features =
   F(FSGSBASE) | F(BMI1) | F(HLE) | F(AVX2) | F(SMEP) |
   F(BMI2) | F(ERMS) | f_invpcid | F(RTM) | f_mpx | F(RDSEED) |
-  F(ADX);
+  F(ADX) | F(SMAP);
 
  /* all calls to cpuid_count() should be made on the same cpu */
  get_cpu();
-- 
1.7.1